### PR TITLE
[LoRA] add a test to ensure `set_adapters()` and attn kwargs outs match

### PR DESCRIPTION
**Pull request by sayakpaul**
Currently, users can set the scale of a loaded LoRA in a pipeline in two ways:

1. `set_adapters(adapter_name, weight)`.
2. Supply something like `attention_kwargs={"scale": weight}` while calling the pipeline.

This PR adds a test to ensure outputs from 1 and 2 match under identical settings.

### CODE REVIEW

File: tests/lora/utils.py [hunk 1]
```diff
@@ -1885,3 +1888,89 @@ def set_pad_mode(network, mode="circular"):
...
+
+            pipe.set_adapters("default", lora_scale)
+            output_lora_scale_wo_kwargs = pipe(**inputs, generator=torch.manual_seed(0))[0]
+            self.assertTrue(
+                not np.allclose(output_no_lora, output_lora_scale_wo_kwargs, atol=1e-3, rtol=1e-3),
+                "Lora + scale should change the output",
+            )
+            self.assertTrue(
+                np.allclose(output_lora_scale, output_lora_scale_wo_kwargs, atol=1e-3, rtol=1e-3),
+                "Lora + scale should change match the output of `set_adapters()`.",
```
Comment (BenjaminBossan) [inline tests/lora/utils.py:1944]:
```suggestion
                "Lora + scale should match the output of `set_adapters()`.",
```

File: tests/lora/utils.py [hunk 2]
```diff
@@ -1885,3 +1888,89 @@ def set_pad_mode(network, mode="circular"):
...
+                self.assertTrue(os.path.isfile(os.path.join(tmpdirname, "pytorch_lora_weights.safetensors")))
+                pipe = self.pipeline_class(**components)
+                pipe = pipe.to(torch_device)
+                pipe.set_progress_bar_config(disable=None)
+                pipe.load_lora_weights(os.path.join(tmpdirname, "pytorch_lora_weights.safetensors"))
+
+                for module_name, module in modules_to_save.items():
+                    self.assertTrue(check_if_lora_correctly_set(module), f"Lora not correctly set in {module_name}")
+
+                print(f"{attention_kwargs=}")
```
Comment (BenjaminBossan) [inline tests/lora/utils.py:1963]: remove

File: tests/lora/utils.py [hunk 3]
```diff
@@ -1885,3 +1888,89 @@ def set_pad_mode(network, mode="circular"):
...
+                    pipe.text_encoder_2.add_adapter(text_lora_config)
+                    self.assertTrue(
+                        check_if_lora_correctly_set(pipe.text_encoder_2), "Lora not correctly set in text encoder 2"
+                    )
+
+            lora_scale = 0.5
+            attention_kwargs = {attention_kwargs_name: {"scale": lora_scale}}
+            output_lora_scale = pipe(**inputs, generator=torch.manual_seed(0), **attention_kwargs)[0]
+            self.assertTrue(
+                not np.allclose(output_no_lora, output_lora_scale, atol=1e-3, rtol=1e-3),
```
Comment (BenjaminBossan) [inline tests/lora/utils.py:1932]: Could use `self.assertFalse`.

**Review by BenjaminBossan — APPROVED** [review]
Thanks, LGTM. Only a couple of nits from my side.

File: tests/lora/utils.py (hunk 2 above)
Comment (sayakpaul) [inline tests/lora/utils.py:1963]:
```suggestion
```

**Review by DN6 — APPROVED** [review]

**Pull request merged**